sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImpl.java

```diff
@@ -66,23 +66,6 @@
     this.flowControlSettings = flowControlSettings;
   }
 
-  @Override
-  @SuppressWarnings("argument.type.incompatible")
-  public void start() throws CheckedApiException {
-    this.subscriber.addListener(this, SystemExecutors.getFuturesExecutor());
-    this.subscriber.startAsync();
-    this.subscriber.awaitRunning();
-    try {
-      this.subscriber.allowFlow(
-          FlowControlRequest.newBuilder()
-              .setAllowedBytes(flowControlSettings.bytesOutstanding())
-              .setAllowedMessages(flowControlSettings.messagesOutstanding())
-              .build());
-    } catch (Throwable t) {
-      throw ExtractStatus.toCanonical(t);
-    }
-  }
-
   private void onMessages(List<SequencedMessage> messages) {
     if (completionFuture.isDone()) {
       return;
```

Comment:
Producing output from another thread will break the assumption that users who use this library may have where they don't need synchronization since methods like startBundle/finishBundle/processElement are all meant to be [accessed from one thread](https://beam.apache.org/documentation/programming-guide/#user-code-thread-compatibility) and will also break similar assumptions in the system (for example counters/monitoring rely on a thread local being set to pass a context object around).

Reply:
This is now fixed.